cli/command/service/rollback.go

```diff
@@ -0,0 +1,67 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/docker/api/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func newRollbackCommand(dockerCli command.Cli) *cobra.Command {
+	options := newServiceOptions()
+
+	cmd := &cobra.Command{
+		Use:   "rollback [OPTIONS] SERVICE",
+		Short: "Rollback a service",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runRollback(dockerCli, cmd.Flags(), options, args[0])
+		},
+		Tags: map[string]string{"version": "1.31"},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.quiet, flagQuiet, "q", false, "Suppress progress output")
+	addDetachFlag(flags, &options.detach)
+
+	return cmd
+}
+
+func runRollback(dockerCli command.Cli, flags *pflag.FlagSet, options *serviceOptions, serviceID string) error {
+	apiClient := dockerCli.Client()
+	ctx := context.Background()
+
+	service, _, err := apiClient.ServiceInspectWithRaw(ctx, serviceID, types.ServiceInspectOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	spec := &service.Spec
+	updateOpts := types.ServiceUpdateOptions{
+		Rollback: "previous",
+	}
+
+	response, err := apiClient.ServiceUpdate(ctx, service.ID, service.Version, *spec, updateOpts)
+
+	if err != nil {
+		return err
+	}
+
+	for _, warning := range response.Warnings {
+		fmt.Fprintln(dockerCli.Err(), warning)
+	}
+
+	fmt.Fprintf(dockerCli.Out(), "%s\n", serviceID)
+
+	if options.detach {
+		warnDetachDefault(dockerCli.Err(), apiClient.ClientVersion(), flags, "rollbacked")
```

Comment:
"rolled back" sounds correct. I checked the context:

"Since --detach=false was not specified, tasks will be rolled back in the background."